### PR TITLE
config_path cleanup

### PR DESCRIPTION
**Pull request by taiyu-len**
a cleanup of `get_config_path` no more temp config nonsense.

guna do a bit more cleanup of config related things later.

### CODE REVIEW

File: sway/config.c [hunk 1]
```diff
@@ -44,113 +47,120 @@ void config_defaults(struct sway_config *config) {
 
 void free_mode(struct sway_mode *mode) {
 	free(mode->name);
-	free_flat_list(mode->bindings);
+	int i;
+	for (i = 0; i < mode->bindings->length; ++i) {
+		struct sway_binding *bind = mode->bindings->items[i];
+		list_free(bind->keys);
```
Comment (SirCmpwn) [inline sway/config.c]: Use free_flat_list here.

File: sway/handlers.c [hunk 2]
```diff
@@ -334,7 +334,6 @@ static bool handle_key(wlc_handle view, uint32_t time, const struct wlc_modifier
 		release_key(sym, key);
 	}
 
-	// TODO: reminder to check conflicts with mod+q+a versus mod+q
```
Comment (SirCmpwn) [inline sway/handlers.c]: Are you sure this works? I'm not.

Comment (taiyu-len) [inline sway/handlers.c]: yes it should,
bindings are sorted from longest to shortest, so longer one have higher priority.
so alt+q+a gets checked before alt+q.
it was this one 470b4dfbae146d83c0061b39534c16b5aad90f1c

Comment (SirCmpwn) [inline sway/handlers.c]: Cool.

File: sway/config.c (hunk 1 above)
Comment (taiyu-len) [inline sway/config.c]: done, also added a bunch of free_**** for each config struct.

**Pull request merged**